dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/PermissionsInterface.java

```diff
@@ -137,6 +137,37 @@ default List<Permission> getPermissionsForWorkflow(User user, Workflow workflow)
      */
     boolean isSharing(User user);
 
+    /**
+     * Returns the userid to use for sharing. The userid will vary by implementation. For SAM
+     * the userid is the user's Google email; the InMemory implementation is the Dockstore username.
+     * If we ever add sharing with GitHub, the username might be the GitHub username, etc.
+     * @param user
+     * @return
+     */
+    Optional<String> userIdForSharing(User user);
+
+    /**
+     * Returns the "original owners" of a workflow as list of {@link Permission}. The original owners
+     * are the <code>workflow.getUsers()</code>, e.g., who created the hosted entry, or members of the
+     * GitHub repo.
+     *
+     * @param workflow
+     * @return
+     */
+    default List<Permission> getOriginalOwnersForWorkflow(Workflow workflow) {
+        return workflow.getUsers().stream()
+            .map(this::userIdForSharing)
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .map(email -> {
+                final Permission permission = new Permission();
+                permission.setEmail(email);
```

Comment:
Does this belong in the Sam implementation of this interface since it assumes the userIdForSharing is an email (and not a github username, etc)

Reply:
The `Permission` class is just a data class; so this will work, but I picked a SAM-specific name for it which is confusing.